### ISSUE TITLE
Make downloading tasks run in parallel

### ISSUE THREAD

**Issue by korenmic** [the-expanse/SideQuest#526]
When 20 updates and 20 new games are downloaded using the sidequest,
they download just one at a time in the tasks view, instead of all (or few) at once.

The parallel downloads feature should be implemented, and there should be added a setting of how many parallel downloads are allowed on the account/sq instance (so it will not be infinite)

that way a few hours of downloads could be finished significantly quicker.

what I usually do to solve this myself is intentionally fail all the tasks so I can copy all the urls at once into an alternative downloading app, it would be great if I could do it within sidequest itself.

In the future it would be nice to also have speed limit (a low priority issue)

**Comment by korenmic** [comment]
by the way, if I fail the downloading tasks, is my account still subscribed to the app for updates?

**Comment by sidequestlegend** [comment]
I havent experienced this issue of downloads being slower in the sidequest application as apposed to in the browser. Indeed parallel downloads could help to mitigate this problem but since it does not affect everyone I would rather try to find the cause and be able to properly work around this. 

Are you using a VPN, or any kind of web filtering software?

**Comment by korenmic** [comment]
> I havent experienced this issue of downloads being slower in the sidequest application as apposed to in the browser. Indeed parallel downloads could help to mitigate this problem but since it does not affect everyone I would rather try to find the cause and be able to properly work around this.
> 
> Are you using a VPN, or any kind of web filtering software?

no, no vpn nor any filtering software. the speed is indeed similar to the browser's speed.
I saw that most of the urls are from github and dropbox, it is pretty normal for them to be relatively slow, but if I download them all at once the downloads' speeds stack and the total speed increase significantly so its confirmed to not be related to anything like a vpn or speed limiting on my side.

**Comment by sidequestlegend** [comment]
ah ok i see so the parallel downloads are all that speed it up, I understand. 

This could take a significant body of work to complete so I couldn't guarantee where it will fall on the road map just yet, but ill keep this issue open so that we can continue the conversation in the mean time. I always agonise over this issue tracker in the run up to a release so its going to get good visibility here. 

Thanks

**Comment by sidequestlegend** [comment]
I have decided this is not really worth while to implement, given the recent news from oculus that they will be handling the download/install process with the new system this feature would not be that useful then .
